src/main/java/com/theelixrlabs/healthcare/service/PatientService.java

```diff
@@ -3,43 +3,89 @@
 import com.theelixrlabs.healthcare.constants.PatientConstants;
 import com.theelixrlabs.healthcare.dao.PatientRepository;
 import com.theelixrlabs.healthcare.dto.PatientDTO;
+import com.theelixrlabs.healthcare.exceptionhandler.CustomException;
 import com.theelixrlabs.healthcare.model.PatientModel;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import java.util.UUID;
 
 /**
  * Service class for managing patient-related operations.
  * Validations on the input parameters are done in this class.
  */
 @Service
 public class PatientService {
-    @Autowired
-    private PatientRepository patientRepository;
+
+    private final PatientRepository patientRepository;
+
+    //Constructor injection of PatientRepository
+    public PatientService(PatientRepository patientRepository) {
+        this.patientRepository = patientRepository;
+    }
+
+    /**
+     * Validates the PatientDTO before adding details.
+     *
+     * @param patientDTO    The data transfer object containing patient information.
+     * @throws CustomException    If validation fails (e.g., empty first name or invalid characters).
+     */
+    private void validatePatientDTO(PatientDTO patientDTO) throws CustomException {
+
+        //Validate first name
+        if (patientDTO.getFirstName().isEmpty()) {
```

Comment:
will it work if field is null ?

Reply:
for null, I am checking using Annotation in PatientDTO .